ludwig/models/trainer.py

```diff
@@ -712,161 +826,91 @@ def train(self, training_set, validation_set=None, test_set=None, save_path="mod
             self.horovod.broadcast_optimizer_state(self.optimizer, root_rank=0)
 
         set_random_seed(self.random_seed)
+
         with training_set.initialize_batcher(
             batch_size=self.batch_size,
             should_shuffle=self.should_shuffle,
             seed=self.random_seed,
             horovod=self.horovod,
         ) as batcher:
-
             # ================ Training Loop ================
-            while progress_tracker.epoch < self.epochs:
+            total_steps = self.epochs * batcher.steps_per_epoch
+
+            # Check save_every_n_steps and adjust if needed.
+            if self.save_every_n_steps == 0 or self.save_every_n_steps > batcher.steps_per_epoch:
```

Comment:
Note that all this is going to be printed on every worker. Usually we only log out messages like this on the coordinator.